makeflow/src/makeflow_module_shared_fs.c

```diff
@@ -0,0 +1,191 @@
+
+#include <assert.h>
+
+#include "debug.h"
+#include "itable.h"
+#include "makeflow_hook.h"
+#include "xxmalloc.h"
+#include "makeflow_log.h"
+#include "makeflow_alloc.h"
+#include "batch_task.h"
+#include "dag.h"
+#include "dag_node.h"
+#include "dag_node_footprint.h"
+#include "dag_file.h"
+#include "jx.h"
+
+static struct list *shared_fs_list = NULL;
+static struct itable *shared_fs_saved_inputs  = NULL;
+static struct itable *shared_fs_saved_outputs = NULL;
+
+/*
+ * Match a filename (/home/fred) to a path stem (/home).
+ * Returns 0 on match, non-zero otherwise.
+ * */
+
+static int prefix_match(void *stem, const void *filename) {
+	assert(stem);
+	assert(filename);
+	return strncmp(stem, filename, strlen(stem));
+}
+
+/*
+ * Returns true if the given filename is located in
+ * a shared filesystem, as given by the shared_fs_list.
+ * */
+
+static int batch_file_on_sharedfs( const char *filename )
+{
+	return !list_iterate(shared_fs_list,prefix_match,filename);
+}
+
+static int create(struct jx *args){
+	shared_fs_list = list_create();
+	shared_fs_saved_inputs  = itable_create(0);
+	shared_fs_saved_outputs = itable_create(0);
+
+	//JX ARRAY ITERATE
+	if(jx_lookup(args, "shared_fs_list")){
+		struct jx *array = jx_lookup(args, "shared_fs_list");
+		struct jx *item = NULL;
+		while((item = jx_array_shift(array))) {
+			list_push_head(shared_fs_list, xxstrdup(item->u.string_value));
```

Comment:
`jx_array_shift()` gives ownership of the items, so you should delete them after use. Also, this will segfault if someone passes in an array with a non-string member.